website/blog/2023-05-26-7.22.0.md

```diff
@@ -0,0 +1,171 @@
+author: Babel Team
+authorURL: https://twitter.com/babeljs
+date:   2022-05-26 0:00:00
+categories: announcements
+share_text: "Babel 7.22.0 Released"
+---
+
+Babel 7.22.0 is out, with transform support for the [Explicit Resource Management](https://github.com/tc39/proposal-explicit-resource-management) proposal parsing, including both the sync and async variants, and with parsing support for the [Import Attributes](https://github.com/tc39/proposal-import-attributes) (an evolution of the old [Import Assertions](http://web.archive.org/web/20220930084518/https://github.com/tc39/proposal-import-assertions) proposal).
+
+We also updated our implementation of [decorators](https://github.com/tc39/proposal-decorators/) following some changes in the proposal, and added support for the TypeScript `import ... =` and `export ... =` statements.
+
+`@babel/preset-env` now includes transform support for the `v` regular expressions flag, which was recently approved as part of the ECMAScript standard, by default. Lastly, we renamed all the plugins for _stable_ ECMAScript features fro `-proposal-` to `-transform-`.
+
+You can read the whole changelog [on GitHub](https://github.com/babel/babel/releases/tag/v7.22.0).
+
+<!-- truncate -->
+
+If you or your company want to support Babel and the evolution of JavaScript, but aren't sure how, you can donate to us on our [Open Collective](https://github.com/babel/babel?sponsor=1) and, better yet, work with us on the implementation of [new ECMAScript proposals](https://github.com/babel/proposals) directly! As a volunteer-driven project, we rely on the community's support to fund our efforts in supporting the wide range of JavaScript users. Reach out at [team@babeljs.io](mailto:team@babeljs.io) if you'd like to discuss more!
+
+## Highlights
+
+### Explicit Resource Management ([#15633](https://github.com/babel/babel/pull/15633), [#15520](https://github.com/babel/babel/pull/15520))
+
+The [explicit resource management](https://github.com/tc39/proposal-explicit-resource-management/) Stage 3 proposal allows definig variables containing resources that will be "disposed" when exiting the scope where they are declared. This is done through `using` (for synchronous disposal) and `await using` (for asynchronous disposal) declarations:
+
+```javascript
+{
+  using fileHandle = filesystem.open("./my/file.txt");
+
+  write(fileHandle, "Hello!");
+} // At the end of the block, fileHandle will be automatically closed.
+```
+
+```javascript
+{
+  await using db = await Database.connect("//my-db.sql");
+  db.query("INSERT INTO tools (name, version) VALUES ('Babel', '7.20.0')");
+} // At the end of the block, the db connection will be closed asynchronously
+```
+
+You can enable support for this proposal by adding `@babel/plugin-proposal-explicit-resource-management` to your Babel config:
+```json title="babel.config.json"
+ {
+   "plugins": [
+     "@babel/plugin-proposal-explicit-resource-management"
+   ]
+ }
+```
```

Comment:
We can also add a try out link:
```
[try out link](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=N4KABGCuDOCWB2BzMAzWAbApgCQIbwBMswBeVDTaAT2gBdMBbAOgHsAHTeACgCImB6BlX5osTWgA9aPAJQBucGADuAJ1j0uonPiKYANGB7ZM6dCwCEshQF8QIUBFxLc6qHCRgCAI1JgnL2jAAEVxaXC9caEwmAGMWeHhMGNpefkEqAFpvJmgAR3QrRWzcyEwVKl4ASQA5AGUAUQAlABUwGuaAeTBaFhZ0aDAueFwGfTAANzK4eJkwADUAQQAZAFV62sGAcgAhcJNNg02AdiYAJlOmAAZNmULrIA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=react&prettier=false&targets=&version=7.22.0&externalPlugins=%40babel%2Fplugin-proposal-explicit-resource-management%407.20.0&assumptions=%7B%7D)
```

The REPL will only enable the react transform and the plugin. The code example is

```js
{
  using fileHandle = filesystem.open("./my/file.txt");
  write(fileHandle, "Hello!");
}

{
  await using db = await Database.connect("//my-db.sql");
  db.query("INSERT INTO tools (name, version) VALUES ('Babel', '7.22.0')");
}
```

The link should be ready as soon as `@babel/standalone` 7.22.0 is published.